homework-submission/week2/src/App.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react";
+import logo from "./logo.svg";
+import todos from "./todos.json";
+import Checkbox from "./Checkbox";
+
+import "./App.css";
+
+class App extends Component {
+  state = {
+    todos
+  };
+
+  onChange = event => {
+    const id = event.target.id;
+    const todos = this.state.todos;
+
+    todos.forEach(todo => {
```

Comment:
Using `forEach` to update array of object is a bad idea. Instead you should use *map*. The reason behind this is because *forEach* will loop over the array and update the todo object itself, however *map* function creates a new array and store the todos with the updated one on this newly created array and return it after the loop end.

Example:-
```
var newTodos = todos.map( todo => {
  if(todo.id === id) {
    return {...todo, done: !todo.done}
  }

return todo;
});

this.setState({ newTodos });
```